src/www/scripts/app.js

```diff
@@ -10,6 +9,23 @@ import SecureStore from "./secure-store";
 import FileStore from "./file-store";
 import LocalStore from "./local-store";
 
+function Emitter() {
+    let events = {};
+    return {
+        on: (eventName, callback) => {
+            events[eventName] = [...(events[eventName] || []), callback];
+            return this;
+        },
+        emit: async (eventName, ...args) => {
+            const callbacks = events[eventName];
```

Comment:
1. If you add a fallback to [] here (as in `|| []`)

Reply:
Done